contracts/Polygon_SpokePool.sol

```diff
@@ -164,10 +171,51 @@ contract Polygon_SpokePool is IFxMessageProcessor, SpokePool {
         _wrap();
     }
 
+    /**
+     * @notice These functions can send an L2 to L1 message so we are extra cautious about preventing a griefing vector
+     * whereby someone batches this call with a bunch of other calls and produces a very large L2 burn transaction.
+     * This might make the L2 -> L1 message fail due to exceeding the L1 calldata limit.
+     */
+
+    function executeUSSRelayerRefundLeaf(
+        uint32 rootBundleId,
+        USSRelayerRefundLeaf memory relayerRefundLeaf,
+        bytes32[] memory proof
+    ) public override {
+        // AddressLibUpgradeable.isContract isn't a sufficient check because it checks the contract code size of
+        // msg.sender which is 0 if called from a constructor function on msg.sender. This is why we check if
+        // msg.sender is equal to tx.origin which is fine as long as Polygon supports the tx.origin opcode.
+        // solhint-disable-next-line avoid-tx-origin
+        if (relayerRefundLeaf.amountToReturn > 0 && msg.sender != tx.origin) revert NotEOA();
+        super.executeUSSRelayerRefundLeaf(rootBundleId, relayerRefundLeaf, proof);
+    }
+
+    function executeRelayerRefundLeaf(
+        uint32 rootBundleId,
+        SpokePoolInterface.RelayerRefundLeaf memory relayerRefundLeaf,
+        bytes32[] memory proof
+    ) public override {
+        // AddressLibUpgradeable.isContract isn't a sufficient check because it checks the contract code size of
+        // msg.sender which is 0 if called from a constructor function on msg.sender. This is why we check if
+        // msg.sender is equal to tx.origin which is fine as long as Polygon supports the tx.origin opcode.
+        // solhint-disable-next-line avoid-tx-origin
+        if (relayerRefundLeaf.amountToReturn > 0 && msg.sender != tx.origin) revert NotEOA();
+        // Prevent calling recipient contract functions atomically with executing relayer refund leaves.
+        _revertIfFunctionCalledAtomically("fill");
```

Comment:
I think this implementation works great, but I have a few minor suggestions:
1. We should avoid using strings, since they are dynamically sized and expensive in solidity. Would suggest using bytes32 for all of these, since bytes32 is fixed size and these can easily fit in that data type.
2. To avoid typos, I would make these compile-time constants at the top of the file like so:
```
    bytes32 private constant FILL_IDENTIFIER = "Fill";
```
and then pass it to the functions -- this prevents string typos from causing the logic to fail.

Reply:
Good call out +1